solr/core/src/java/org/apache/solr/cluster/placement/impl/CollectionMetricsBuilder.java

```diff
@@ -61,18 +80,19 @@ public ShardMetrics build() {
       replicaMetricsBuilders.forEach((name, replicaBuilder) -> {
         ReplicaMetrics metrics = replicaBuilder.build();
         metricsMap.put(name, metrics);
-        // skip leader from map
         if (replicaBuilder.leader) {
           if (leaderMetricsBuilder == null) {
             leaderMetricsBuilder = replicaBuilder;
           }
-          if (replicaBuilder != leaderMetricsBuilder) {
```

Comment:
Why did you remove that check? It doesn't deserve a new if statement (just `else` to the `leaderMetricsBuilder == null` condition is equivalent) but it's interesting, it means two replicas pretend they are shard leader... Maybe error log?